__test__/integration/servicing_api/runners/get_total_expected_repayment.ts

```diff
@@ -0,0 +1,113 @@
+// libraries
+import * as ABIDecoder from "abi-decoder";
+import * as moment from "moment";
+import * as Web3 from "web3";
+import { BigNumber } from "../../../../utils/bignumber";
+
+// utils
+import * as Units from "../../../../utils/units";
+
+import { AdaptersAPI, ContractsAPI, OrderAPI, ServicingAPI, SignerAPI } from "../../../../src/apis";
+import { DebtOrder } from "../../../../src/types";
+import {
+    DummyTokenContract,
+    RepaymentRouterContract,
+    TokenTransferProxyContract,
+} from "../../../../src/wrappers";
+
+import { ACCOUNTS } from "../../../accounts";
+
+const web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+
+const contractsApi = new ContractsAPI(web3);
+const adaptersApi = new AdaptersAPI(web3, contractsApi);
+const orderApi = new OrderAPI(web3, contractsApi, adaptersApi);
+const signerApi = new SignerAPI(web3, contractsApi);
+const servicingApi = new ServicingAPI(web3, contractsApi);
+
+const TX_DEFAULTS = { from: ACCOUNTS[0].address, gas: 400000 };
+
+import { GetTotalExpectedRepaymentScenario } from "../scenarios";
+
+export class GetTotalExpectedRepaymentRunner {
+    public static testScenario(scenario: GetTotalExpectedRepaymentScenario) {
+        let principalToken: DummyTokenContract;
+        let tokenTransferProxy: TokenTransferProxyContract;
+        let repaymentRouter: RepaymentRouterContract;
+        let debtOrder: DebtOrder.Instance;
+        let issuanceHash: string;
+
+        const CONTRACT_OWNER = ACCOUNTS[0].address;
+
+        const CREDITOR = ACCOUNTS[1].address;
+
+        const DEBTOR = ACCOUNTS[2].address;
+
+        beforeAll(async () => {
+            const tokenRegistry = await contractsApi.loadTokenRegistry();
+            const principalTokenAddress = await tokenRegistry.getTokenAddressBySymbol.callAsync(
+                "REP",
+            );
+
+            repaymentRouter = await contractsApi.loadRepaymentRouterAsync();
+
+            tokenTransferProxy = await contractsApi.loadTokenTransferProxyAsync();
```

Comment:
shouldn't be necessary to load the `tokenTransferProxy` -- we can use the`tokenApi.setUnlimitedProxyAllowance` endpoint instead of having to fiddle with the raw contracts.  Given that this is an integration test, that would be preferrable IMO